src/bartiq/compilation/_evaluate.py

```diff
@@ -119,7 +119,20 @@ def _parse_assignments(routine: Routine, assignments: list[str]) -> list[Assignm
     # Sort the assignments based on whether they refer to register size variables or not
     parsed_assignments: list[Assignment] = []
     for variable, value_str in assignment_map.items():
-        value = parse_value(value_str)
+        try:
+            value = parse_value(value_str)
+        except BartiqCompilationError:
+            # str to expression
+            expression = sympy_backend.as_expression(value_str)
```

Comment:
[issue] It should be using `backend` and not `sympy_backend`. The only reason `sympy_backend` is in this file is because we assign it as a default backend.